### PR TITLE
sink (ticdc): fix a deadlock due to checkpointTs fall back in sinkNode

### LINKED ISSUE #4055: changefeed get stuck in owner switch case
### What did you do?

- 5 captures, 1 changefeed, downstream is mysql
- run CDC owner switch case in test infra, ( tpcc run parallelize with owner switch)

### What did you expect to see?

no error

### What did you see instead?

- changefeed get stuck
- changefeed can be paused or resumed

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
5.0.6
```

TiCDC version (execute `cdc version`):

```console
Release Version: v5.0.6
Git Commit Hash: eba76b8952647d1cb648031f0f505e4d7ff91acf
Git Branch: heads/refs/tags/v5.0.6
UTC Build Time: 2021-12-23 12:26:43
Go Version: go version go1.13 linux/amd64
Failpoint Build: false
```

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #4055

### What is changed and how it works?
The code changed is simple and the reason is described in comments.	

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
Fix a bug that can cause changefeed stuck due to a deadlock occurs.
```

### CODE REVIEW

File: cdc/processor/pipeline/sink.go [hunk 1]
```diff
@@ -154,12 +154,21 @@ func (n *sinkNode) flushSink(ctx context.Context, resolvedTs model.Ts) (err erro
...
+
+	// we must call flowController.Release immediately after we call
+	// FlushRowChangedEvents to prevent deadlock cause by checkpointTs
+	// fall back
+	n.flowController.Release(checkpointTs)
+
+	// the checkpointTs may fall back in some situation such as:
+	//   1. This table is newly added to the processor
+	//   2. There is one table in the processor that has a smaller
+	// checkPointTs than this one
```
Comment (overvenus) [inline cdc/processor/pipeline/sink.go:166]:
```suggestion
	//   checkpointTs than this one
```

**Review by overvenus — COMMENTED** [review]
Rest LGTM

**Review by overvenus — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]
LGTM

**Pull request merged**